src/storage/buffer_manager/buffer_manager.cpp

```diff
@@ -193,25 +199,35 @@ void handleAccessViolation(unsigned int exceptionCode, PEXCEPTION_POINTERS excep
 
 // Returns true if the function completes successfully
 inline bool try_func(const std::function<void(uint8_t*)>& func, uint8_t* frame,
-    const std::vector<std::unique_ptr<VMRegion>>& vmRegions, PageSizeClass pageSizeClass) {
-#if defined(_WIN32)
+    const std::array<std::unique_ptr<VMRegion>, 2>& vmRegions [[maybe_unused]],
+    PageSizeClass pageSizeClass [[maybe_unused]]) {
+#if BM_MALLOC
+    if (frame == nullptr) {
+        return false;
+    }
+#endif
+
+#if defined(_WIN32) && !BM_MALLOC
```

Comment:
The macro here is becoming more a bit hard to follow now. I wonder if we should choose to duplicate the code a bit to separate them more clearly.